src/StreamJsonRpc/Reflection/TargetMethod.cs

```diff
@@ -97,6 +101,7 @@ internal object Invoke(CancellationToken cancellationToken)
                 this.arguments[this.arguments.Length - 1] = cancellationToken;
             }
 
+            this.dispatchEvents?.OnMethodInvoking(this.request, this.signature.MethodInfo, this.target, cancellationToken);
```

Comment:
I don't think this is the right layer to make this call. For one thing, as mentioned in another comment this callback needs to support async implementations for our use case of making another service call to check authorization and this `TargetMethod.Invoke` method is not async already (but its caller is).
But as you'll notice in the caller, it's *crucial* that we invoke this method before any other yielding awaits in that calling method in order to preserve the message order preservation guarantee which almost everyone assumes is present and many people rely on. This makes adding an async pre-step before invoking the RPC method more difficult to imagine how to implement correctly.